cg/store/filters/status_illumina_flow_cell_filters.py

```diff
@@ -0,0 +1,32 @@
+"""Fillters for the Illumina Flow Cells."""
+
+from enum import Enum
+
+from sqlalchemy.orm import Query
+
+from cg.store.models import IlluminaFlowCell
+
+
+def filter_illumina_flow_cell_by_internal_id(
+    flow_cells: Query, internal_id: str, **kwargs
+) -> Query:
+    """Get an Illumina Flow Cell by its internal id."""
+    return flow_cells.filter_by(internal_id=internal_id)
+
+
+def apply_illumina_flow_cell_filters(
+    flow_cells: Query, filter_functions: list[callable], internal_id: str
+) -> Query:
+    """Apply filtering functions and return filtered results."""
+    for function in filter_functions:
+        flow_cells: Query = function(
+            flow_cells=flow_cells,
+            internal_id=internal_id,
+        )
+    return flow_cells
+
+
+class IlluminaFlowCellFilter(Enum):
+    """Define FlowCell filter functions."""
+
+    BY_INTERNAL_ID: callable = filter_illumina_flow_cell_by_internal_id
```

Comment:
All devices will have an internal ID we can filter on. Does the filter class need to be called Illumina_something?

Reply:
Right now it would clash with the normal "flow cell" filters. We can refactor once we are on the new flow.